src/plugins/jni/jni.c

```diff
@@ -321,8 +321,14 @@ int elektraJniClose (Plugin * handle, Key * errorKey)
 {
 	Data * data = elektraPluginGetData (handle);
 
-	if (!data || data->module == 1)
+	if (!data)
+	{
+		return 0;
+	}
+
+	if (data->module == 1)
 	{
+		elektraFree (data);
```

Comment:
@markus2330 this is one instance where I think it was not cleaned up properly. The leak was triggered when Elektra searched for the best provider for `binary`, which should be `base64`.

In any case: do you agree with the changes?